### DISCUSSION TITLE
Retorno do CodMun pelo SPI para Validação do Pagamento

### DISCUSSION THREAD

**Discussion by mvmarcal** [bacen/pix-api#320] [Q&A] [answered]
Olá,

Para pagamentos com Vencimento, o **_PSP do Recebedor_** precisa receber o CodMun para realizar os calculos conforme dias úteis do município do **_Usuário Pagador_**.

Sabendo que mais de uma pessoa pode realizar a leitura de um QR Code, podemos receber leituras de diversos municípios e estas podem ter valores diferentes.

Considerando o cenário onde para uma mesma cobrança (txid = cob1):

> 1. Pagador 1 realiza a leitura do QR Code com CodMun=Rio de Janeiro, para pagamento imediato, onde ganha 5% de desconto
> 2. Pagador 2 realiza a leitura do QR Code com CodMun=São Paulo, agendando o pagamento para 10 dias úteis depois, para fins de consulta, onde não terá desconto
> 3. Pagador 1 Efetiva o Pix

No fluxo _"3.1. Fluxo de transações entre participantes diretos"_ do "[III.ManualdeFluxosdoProcessodeEfetivacaodoPix-versao1.1.pdf](https://www.bcb.gov.br/content/estabilidadefinanceira/pix/Regulamento_Pix/III.ManualdeFluxosdoProcessodeEfetivacaodoPix-versao1.1.pdf)", Step 8 onde o **_PSP do Recebedor_** recebe a msg de crédito, também será recebido o CodMun?
Pois será necessário para o **_PSP do Recebedor_** validar se o valor recebido está correto, conforme a leitura realizada do _**Pagador 1**_

**Comment by ninrod** [comment, marked as answer]
Olá @mvmarcal, 

Essa questão já apareceu antes. 

Primeiro, você teria que estabelecer um exemplo mais concreto; isso porque não é permitido variar o valor da cobrança em função do codmun (dar 5% de desconto só porque o pagador está no rio de janeiro), mas sim aplicar feriados locais para postergar, se aplicável, a data de vencimento da cobrança para o próximo dia útil efetivo.

Em segundo lugar, sugiro você ler a [minha resposta aqui](https://github.com/bacen/pix-api/issues/199#issuecomment-733066883), na issue #199. Aliás, ler a issue inteira é interessante.

**Reply by renatofrota** [reply]
Ele associou o desconto de 5% ao fato do pagamento ser imediato, não à localidade.

**Reply by ninrod** [reply]
Ah, ok.

**Reply by mvmarcal** [reply]
Me desculpe, talvez o cenário não expressou o que eu gostaria.
Segue um cenário correto:
- > Cosiderando uma mesma cobrança (txid = cob1) de R$100,00
- > Cobrança venceu 10 dias corridos atrás e ainda está válida
- > Juros da cobrança é de 1% ao dia
- > Pagador 1 realiza a leitura do QR Code com CodMun=Rio de Janeiro, para pagamento imediato, onde se passaram 6 dias úteis
- > Pagador 2 realiza a leitura do QR Code com CodMun=São Paulo, para pagamento imediato, onde se passaram 7 dias úteis devido à um feriado municipal
- > Pagador 1 Efetiva o Pix pois para este está mais barato devido à menos dias de juros
- > Neste cenário para conferir o valor, será necessário o Codmun

Para isso, a SPI irá retornar o Codmun?

**Reply by ninrod** [reply]
Não. Para isso, cabe ler a minha resposta: https://github.com/bacen/pix-api/issues/199#issuecomment-733066883

**Reply by mvmarcal** [reply]
Obrigado pela resposta!
